### PR TITLE
tree-wide: check for NULLs in more places

### PR DESCRIPTION
**Pull request by mrc0mmand**
Fixes issues pointed out by the `cpp/inconsistent-null-check` LGTM
query.

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: src/analyze/analyze.c [hunk 1]
```diff
@@ -982,7 +982,7 @@ static int list_dependencies_one(sd_bus *bus, const char *name, unsigned level,
 
         STRV_FOREACH(c, deps) {
                 times = hashmap_get(unit_times_hashmap, *c);
-                if (times_in_range(times, boot) && times->activated >= service_longest)
+                if (times && times_in_range(times, boot) && times->activated >= service_longest)
```
Comment (keszybz) [inline src/analyze/analyze.c]: times_in_range() checks for NULL… So all three checks that are added here don't matter.

Comment (mrc0mmand) [reply src/analyze/analyze.c]: Hah, I see, thanks. I'll suppress the warning in these cases.

File: src/analyze/analyze.c [hunk 2]
```diff
@@ -981,7 +981,7 @@ static int list_dependencies_one(sd_bus *bus, const char *name, unsigned level,
                 return r;
 
         STRV_FOREACH(c, deps) {
-                times = hashmap_get(unit_times_hashmap, *c);
+                times = hashmap_get(unit_times_hashmap, *c); /* lgtm [cpp/inconsistent-null-check] */
```
Comment (bluca) [inline src/analyze/analyze.c]: is it possible to mark it on the service's database instead, like on coverity?

Comment (mrc0mmand) [reply src/analyze/analyze.c]: I guess we could just use the "hide from the results" function (in both LGTM and CodeQL UIs).

Comment (bluca) [reply src/analyze/analyze.c]: yeah, sounds more palatable than annotating the code directly, given it's parser-specific

**Pull request merged**